### PR TITLE
chore: add tests for CreateNamespace and DeleteNamespace in jobs.go

### PR DESCRIPTION
**Pull request by salonig23**
## Ticket

DET-10436

## Description

Added tests for CreateNamespace and DeleteNamespace. Ended up splitting CreateNamespace into a helper function because testing syncNamespaces proved to be very tedious due to mocking - and eventually reached a roadblock where I wasn't able to mock Event.Watch()

## Test Plan

CI passes

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/rm/kubernetesrm/kubernetes_resource_manager_intg_test.go [hunk 1]
```diff
@@ -851,6 +853,142 @@ func TestVerifyNamespaceExists(t *testing.T) {
 	require.Error(t, err)
 }
 
+func TestCreateNamespaceHelper(t *testing.T) {
```
Comment (amandavialva01) [inline master/internal/rm/kubernetesrm/kubernetes_resource_manager_intg_test.go:856]: I think it's ok to call this `TestCreateNamespace` even though it only implicitly calls the helper!

File: master/internal/rm/kubernetesrm/kubernetes_resource_manager_intg_test.go [hunk 2]
```diff
@@ -851,6 +853,142 @@ func TestVerifyNamespaceExists(t *testing.T) {
...
+	js := createMockJobsService(nil, device.CPU, false)
+
+	validNamespace := "valNamespace"
+	existentNamespace := "nonexistentNamespace"
```
Comment (amandavialva01) [inline master/internal/rm/kubernetesrm/kubernetes_resource_manager_intg_test.go:860]: I think we should rename this var `nonExistentNamespace`

File: master/internal/rm/kubernetesrm/kubernetes_resource_manager_intg_test.go [hunk 3]
```diff
@@ -851,6 +853,142 @@ func TestVerifyNamespaceExists(t *testing.T) {
...
+		require.True(t, ok)
+		_, ok = worker.jobInterface[validNamespace]
+		require.True(t, ok)
+		_, ok = worker.configMapInterfaces[validNamespace]
+		require.True(t, ok)
+	}
+
+	// test with erroneous namespace
+	err = js.createNamespace(erroneousNamespace)
+	require.ErrorContains(t, err, "random error")
```
Comment (amandavialva01) [inline master/internal/rm/kubernetesrm/kubernetes_resource_manager_intg_test.go:956]: Can we make "ranom error" into a string and pass that into the mockRM return statement and here?

File: master/internal/rm/kubernetesrm/kubernetes_resource_manager_intg_test.go [hunk 4]
```diff
@@ -851,6 +853,142 @@ func TestVerifyNamespaceExists(t *testing.T) {
...
+	err := js.deleteNamespace(validNamespace)
+	require.NoError(t, err)
+
+	// test with non-existent namespace
+	err = js.deleteNamespace(nonexistentNamespace)
+	require.NoError(t, err)
+
+	// test with erroneous namespace
+	err = js.deleteNamespace(erroneousNamespace)
+	require.ErrorContains(t, err, "random error")
```
Comment (amandavialva01) [inline master/internal/rm/kubernetesrm/kubernetes_resource_manager_intg_test.go:989]: Same as my previous comment with the "random error"

**Review by amandavialva01 — APPROVED** [review]
This LGMT modulo a few small nits, nice tests!

File: master/internal/rm/kubernetesrm/kubernetes_resource_manager_intg_test.go (hunk 2 above)
Comment (salonig23) [reply master/internal/rm/kubernetesrm/kubernetes_resource_manager_intg_test.go]: done

File: master/internal/rm/kubernetesrm/kubernetes_resource_manager_intg_test.go (hunk 3 above)
Comment (salonig23) [reply master/internal/rm/kubernetesrm/kubernetes_resource_manager_intg_test.go]: done

**Review by kkunapuli — COMMENTED** [review]
LGTM!

**Pull request merged**